drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -64,15 +80,24 @@ static int stm32_temp_channel_get(const struct device *dev, enum sensor_channel
 		return -ENOTSUP;
 	}
 
-	int32_t mv = data->raw * cfg->tsv_mv / 0x0FFF; /* Sensor value in millivolts */
-
-	if (cfg->is_ntc) {
-		temp = (float)(cfg->v25_mv - mv);
+	if (cfg->has_cal) {
```

Comment:
I was indeed thinking of using more the defines, but that would make the code either work compile time for one or the other compatible. While now it's "ready" to work for more than one compatible at the same time (when reworking DRV_COMPAT in driver config creations ). 
On the other hand you can argue if there would ever be more than 1 temp sensor in the mcu itself... so you have a preference for this?

Reply:
Both hardware blocks are mutually exclusive, so let's take it into consideration and use this to get a smaller footprint driver.
(Otherwise current proposal was indeed a good way to handle differences in runtime)

Reply:
That's clear, thanks for the answer. I've cleaned it up a bit with this info. Also made only 1 instance possible, as it doesn't make sense, and it adds up to the readability